File: hacker_rank/30_days/day15/day15.c
#include <stdlib.h>
#include <stdio.h> 

typedef struct Node
{
    int data;
    struct Node* next;
}Node;

Node* insert(Node *head,int data)
{
    Node* new_node = malloc(sizeof(Node));

    Node* tail = head;
    while(tail->next != NULL) {
      tail = tail->next;
    }

    tail->next = new_node;
    new_node->data = data;
    new_node->next = NULL;
    return head;
}

void display(Node *head)
{
  Node *start=head;
  while(start)
  {
    printf("%d ",start->data);
    start=start->next;
  }
}

int main()
{
  int T,data;
    scanf("%d",&T);
    Node *head=NULL;  
    while(T-->0){
        scanf("%d",&data);
        head=insert(head,data);
    }
    display(head);
}

File: hacker_rank/30_days/day15/day15.py
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None
class Solution:
    def display(self,head):
        current = head
        while current:
            print current.data,
            current = current.next

    def insert(self,head,data):
        
        if head == None:
            new_node = Node(data)
            head = new_node
            return head
        
        next_node = head
        while next_node.next != None:
            next_node = next_node.next
        new_node = Node(data)
        next_node.next = new_node
        return head

mylist= Solution()
T=int(input())
head=None
for i in range(T):
    data=int(input())
    head=mylist.insert(head,data)    
mylist.display(head);